app/decorators/notification_decorator.rb

```diff
@@ -45,6 +45,49 @@ def to_model
     self
   end
 
+  def subscription_for(user)
+    subscription_to_comment_ancestry_for(user) ||
+      subscription_to_comment_for(user) ||
+      subscription_to_commentable_article_for(user) ||
+      subscription_to_article_for(user)
```

Comment:
So, _particularly_ for folks who reviewed this PR previously (but also for anyone/everyone following along) I just want to call out that this is the biggest functional shift to getting this to work. 

First: `comment_ancestry` is a _lot_ more involved than had been previously reckoned with in this branch. It's not _just_ `comment.id.to_s` — it is a **join** of _all_ the comment's parents ("6/7/8/9/etc"). 

To re-iterate from another comment: we can't really _leap_ from the `ancestry` to the "this is a subscribe-to-thread" scenario. And we can't treat `ancestry` as if it's just an id. The user _might_ be subscribed to the comment's parent, or grand-parent, or great-grand-parent (or might be subscribed to the whole article, or might have once been subscribed and no longer, etc...)

I'm also a little worried that this is going to have some serious performance implications — the notifications page is long and this is potentially a _lot_ of extra queries. (It's not really as if _every_ line here will lead to a query, but there's likely at least two per comment notification.) The performance isn't noticably bad locally (even with _lots_ of notifications loaded) but I recommend we try rolling the feature flag out in gradual increments.